pkg/apis/autoscaling/v1alpha1/metricsproducer_validation.go

```diff
@@ -12,40 +12,61 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// +kubebuilder:webhook:verbs=create;update,path=/validate-autoscaling-karpenter-sh-v1alpha1-metricsproducer,mutating=false,sideEffects=None,failurePolicy=fail,groups=autoscaling.karpenter.sh,resources=metricsproducers,versions=v1alpha1,name=vmetricsproducer.kb.io
```

Comment:
Did you try playing with more sane paths here? 

/validate-autoscaling-karpenter-sh-v1alpha1-metricsproducer -> autoscaling/v1alpha1/metricsproducer/validate

Reply:
https://github.com/kubernetes-sigs/kubebuilder/issues/1333
Unable to change the path with kubebuilder. I tried, and it just doesn't change. If we stop using KB it should work though. I might do this in a future PR.